realm/realm-library/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -1044,4 +1045,123 @@ public void onChange(AllTypes object) {
             }
         });
     }
+
+    @Test
+    @RunTestInLooperThread
+    public void realmListener_realmResultShouldBeSynced() {
+        final AtomicBoolean changedOnce = new AtomicBoolean(false);
+        final Realm realm = looperThread.realm;
+        realm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                realm.createObject(AllTypes.class);
+            }
+        });
+        final RealmResults<AllTypes> results = realm.where(AllTypes.class).findAll();
+        assertEquals(1, results.size());
+
+        realm.addChangeListener(new RealmChangeListener<Realm>() {
+            @Override
+            public void onChange(Realm element) {
+                if (!changedOnce.get()) {
+                    // Change event triggered by populating
+                    assertEquals(1, realm.where(AllTypes.class).count());
+                    assertEquals(1, results.size());
+
+                    realm.executeTransactionAsync(new Realm.Transaction() {
+                        @Override
+                        public void execute(Realm realm) {
+                            AllTypes allTypes = realm.where(AllTypes.class).findFirst();
+                            assertNotNull(allTypes);
+                            allTypes.deleteFromRealm();
+                            final RealmResults<AllTypes> results = realm.where(AllTypes.class).findAll();
+                            assertEquals(0, results.size());
+                        }
+                    });
+                    changedOnce.set(true);
+                } else {
+                    // Change event triggered by deletion in async transaction.
+                    assertEquals(0, realm.where(AllTypes.class).count());
+                    assertEquals(0, results.size());
+                    realm.close();
+                    looperThread.testComplete();
+                }
+            }
+        });
+    }
+
+    // We precisely depend on the order of triggering change listeners right now.
+    // So it should be:
+    // 1. Synced object listener
```

Comment:
I know, my comment was mostly that they where missing from your 1.2.3 list. So maybe just add a comment about that?